runtime/tests/integration_tests.rs

```diff
@@ -895,3 +898,108 @@ fn test_idev_id_cert() {
         caliptra_drivers::CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS.into()
     );
 }
+
+#[test]
+fn test_extend_pcr_cmd() {
+    fn generate_mailbox_extend_pcr_req(
+        pcr_idx: u32,
+        data_size: usize,
+        payload: [u8; ExtendPcrReq::DATA_MAX_SIZE],
+    ) -> Result<ExtendPcrReq, ExtendPcrReqErr> {
+        let cmd = ExtendPcrReq::new(
+            MailboxReqHeader { chksum: 0 },
+            pcr_idx,
+            data_size as u32,
+            payload,
+        )?;
+
+        let checksum = caliptra_common::checksum::calc_checksum(
+            u32::from(CommandId::EXTEND_PCR),
+            &cmd.as_bytes()[4..],
+        );
+
+        ExtendPcrReq::new(
+            MailboxReqHeader { chksum: checksum },
+            cmd.pcr_idx,
+            cmd.data_size,
+            cmd.data,
+        )
+    }
+
+    // Testing for payload_data [0,...,0]
+    let mut model = run_rt_test(None, None);
+    let payload_data = [0u8; ExtendPcrReq::DATA_MAX_SIZE];
+
+    let cmd = generate_mailbox_extend_pcr_req(4, payload_data.len(), payload_data).unwrap();
+    let res = model.mailbox_execute(u32::from(CommandId::EXTEND_PCR), &cmd.as_bytes());
+    assert!(res.is_ok());
+
+    // Testing for high entropy test vector
+    let mut model = run_rt_test(None, None);
+    let payload_data: [u8; ExtendPcrReq::DATA_MAX_SIZE] = [
+        35, 132, 102, 150, 202, 72, 140, 224, 37, 129, 164, 104, 69, 198, 15, 5, 160, 64, 241, 233,
+        110, 136, 79, 238, 246, 146, 166, 165, 129, 108, 40, 15, 155, 76, 138, 220, 122, 239, 26,
+        182, 215, 239, 9, 244, 123, 192, 213, 72, 247, 223, 97, 50, 203, 98, 85, 255, 185, 136,
+        117, 132, 45, 47, 144, 171, 30, 11, 237, 104, 239, 127, 74, 45, 22, 202, 113, 89, 231, 46,
+        208, 185, 237, 115, 123, 223, 231, 246, 254, 103, 112, 39, 33, 159, 23, 201, 17, 64, 144,
+        24, 179, 34, 251, 43, 5, 215, 169, 71, 118, 156, 103, 123, 101, 170, 5, 38, 190, 72, 75,
+        81, 223, 72, 131, 215, 100, 50, 150, 57, 34, 255, 172, 39, 211, 121, 252, 89, 24, 57, 15,
+        255, 175, 12, 201, 179, 23, 247, 117, 169, 254, 227, 206, 220, 210, 169, 217, 42, 172, 102,
+        146, 234, 16, 51, 40, 9, 37, 177, 251, 80, 134, 0, 113, 8, 244, 140, 23, 133, 119, 94, 216,
+        38, 247, 184, 174, 36, 0, 78, 151, 201, 134, 182, 124, 122, 105, 75,
+    ];
+
```

Comment:
I don't think this is the mailbox API is intended. You want to pass a digest of data, not the data itself. The mailbox itself is limited in size and PCRs often get extended with digests of code much larger than the mailbox itself.